sound/soc/sof/loader.c

```diff
@@ -129,6 +130,70 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 }
 EXPORT_SYMBOL(snd_sof_fw_parse_ext_data);
 
+/* parse extended FW manifest data structures */
+static int snd_sof_fw_xman_parse(struct snd_sof_dev *sdev, const void *data)
+{
+	struct xman_header *head = (struct xman_header *)data;
+	struct xman_elem_header *elem_hdr;
+	u32 remaining = head->size - head->header_size;
+	uintptr_t iptr;
+	int ret = 0;
+
+	if (XMAN_VERSION_INCOMPATIBLE(XMAN_VERSION, head->version))
+		dev_warn(sdev->dev, "warning: extended manifest version 0x%X differ from used 0x%X\n",
+			 head->version, XMAN_VERSION);
+
+	/* get first header */
+	iptr = (uintptr_t)data + head->header_size;
+	elem_hdr = (struct xman_elem_header *)iptr;
+
+	while (remaining > 0) {
+		dev_dbg(sdev->dev, "found ext header type %d size 0x%x\n",
+			elem_hdr->type, elem_hdr->size);
+
+		/* process structure data */
+		switch (elem_hdr->type) {
+		default:
```

Comment:
oh man this is horrible. you have a useless check here?

Reply:
rather not

Reply:
To fully introduce extended manifest mechanism separately from introducing extended manifest elements in following commits